core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -583,8 +588,29 @@ public UpdateNode plan(Update node)
         PlanAndMappings planAndMappings = coerce(builder, orderedColumnValues, analysis, idAllocator, symbolAllocator, typeCoercion);
         builder = planAndMappings.getSubPlan();
 
+        TableMetadata tableMetadata = metadata.getTableMetadata(session, handle);
```

Comment:
The planner shouldn't be accessing table metadata directly or trying to match column names (there are potential case sensitivity issues that need to be handled at analysis time) -- this information should be recorded during analysis and used here instead.  (I know the existing code is already doing some of that, but that should be cleaned up first)

Also, the proposed code is now calling both `getTableSchema()` and `getTableMetadata()`, which is unnecessary. `getTableMetadata()` returns a superset of what `getTableSchema()` returns, so we should reuse that and avoid the extra metadata call.

Reply:
@martint thanks for your input. So will it be ok if I just fix that second point - call metadata only once? I actually noticed that but didn't want to make too big of a change.

Reply:
Yes, that's ok, but please add a TODO and/or file an issue to clean up that metadata access logic (and move it to the analyzer) so that we don't lose track of it.

Reply:
Sure, will do, last thing, I know that similar thing was done in LogicalPlanner here https://github.com/trinodb/trino/blob/adcfbe95250f216170046d1ab8fb108b61d1c61e/core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java#L447 Should this also be moved to analyzer?

Reply:
Yes